api/src/main/java/au/org/aodn/nrmn/restapi/service/SurveyEditService.java

```diff
@@ -0,0 +1,150 @@
+import static au.org.aodn.nrmn.restapi.util.TimeUtils.parseTime;
+
+@Service
+public class SurveyEditService {
+
+    @Autowired
+    SurveyRepository surveyRepository;
+
+    @Autowired
+    SiteRepository siteRepository;
+
+    @Autowired
+    DiverRepository diverRepository;
+
+    @Autowired
+    ModelMapper mapper;
+
+    public Survey updateSurvey(SurveyDto surveyDto) {
+
+        Survey survey = surveyRepository.findById(surveyDto.getSurveyId()).orElseThrow(ResourceNotFoundException::new);
+
+        mapper.map(surveyDto, survey);
+
+        Site site = siteRepository.findBySiteCode(surveyDto.getSiteCode());
+        survey.setSite(site);
+
+        Diver pqDiver = diverRepository.findByCriteria(surveyDto.getPqDiverInitials()).get(0);
+        survey.setPqDiverId(pqDiver.getDiverId());
+
+        return survey;
+    }
+
+    public ValidationErrors validateSurvey(SurveyDto surveyDto) {
+        List<ValidationError> errors = new ArrayList<>();
+        Site surveyDtoSite = siteRepository.findBySiteCode(surveyDto.getSiteCode());
+        Date surveyDate = null;
+
+        // Date validations
+        try {
+            surveyDate = parseDate(surveyDto.getSurveyDate());
+        } catch (DateTimeException e) {
+            errors.add(new ValidationError("Survey", "surveyDate", surveyDto.getSurveyDate(),
+                    e.getMessage()));
+        }
+
+        if(surveyDate != null) {
+            if (surveyDate.after(new Date())) {
+                errors.add(new ValidationError("Survey", "surveyDate", surveyDto.getSurveyDate(),
+                        "A survey date cannot be in the future."));
+            }
+
+            if (surveyDate.before(Date.from(LocalDate.of(2006, 1, 1).atStartOfDay(ZoneId.systemDefault()).toInstant()))) {
+                errors.add(new ValidationError("Survey", "surveyDate", surveyDto.getSurveyDate(),
+                        "A survey date cannot be before January 1st, 2006."));
+            }
+        }
+
+        // Time validations
+        try {
+            parseTime(surveyDto.getSurveyTime());
+        } catch (DateTimeException e) {
+            errors.add(new ValidationError("Survey", "surveyTime", surveyDto.getSurveyTime(),
+                    "The survey time must be in the format hh:mm[:ss]"));
+        }
+
+        // Site code and site name exist and match
+        if (!surveyDtoSite.getSiteName().equals(surveyDto.getSiteName())) {
+            errors.add(new ValidationError("Site", "siteCode", surveyDto.getSiteName(),
+                    String.format("The site code %s does not match the site name \"%s\"", surveyDto.getSiteCode(), surveyDto.getSiteName())));
+        }
+
+        // Lat and Lon match site
+        String lat = surveyDto.getLatitude();
+        if(!isEmpty(lat) && !truncateNumber(surveyDtoSite.getLatitude()).equals(truncateNumber(Double.parseDouble(lat)))){
+            errors.add(new ValidationError("Survey", "latitude", surveyDto.getSiteName(),
```

Comment:
This should be `lat` instead of `surveyDto.getSiteName()`

Reply:
Whoops